tests/functions/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -170,4 +190,27 @@ function gen(): iterable
   yield 3;
 }
 
+$db->Execute($sql, [
+  $foo,
+  $bar,
+  $foobar,
+  $somewhatLongParameter,
+  $evenLongerParameter
+]);
+
+$app->get('/hello/{name}', function ($name) use ($app) {
+  return 'Hello ' . $app->escape($name);
+});
+
+$this->something->method($argument, $this->more->stuff($this->even->more->
+  things->
+  complicatedMethod()));
```

Comment:
I tried using a `conditionalGroup` to wrap function calls just like the JS printer does - which works generally quite well, but I'm still having issues with this nested call example. Maybe @azz can point me in the right direction?

Reply:
@czosel do you know where the JS printer does this? I'm mostly just curious

Reply:
@mgrip https://github.com/prettier/prettier/blob/master/src/language-js/printer-estree.js#L3195

Reply:
🙏

Reply:
JS actually just does the long form in that example. I'm beginning to wonder if we're actually going to be able to print everything in the way PSR12 outlines

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEMAWBLAzgOixAWznQygHMcj0IATACgEMAnMgVyNgBoACE3AiEzh4YrAGZi6AHSjdemXHABuCSoOElyuSAQAOAGwxgG8GgFliaWnQCUMmzYDcITiAi6YGaFmShmTCAB3AAVmBB8UBiUIDBoXEAYsGGQxBn0sOFcAIyYGMABrYgBlXTzSMmQYJlZMkBoIMBS0jNdSDKYYYNyyAgYm9NqAKywADwAhXILihiIAGVI4fpa3VhhdVYAmJdrSpnbkECyGLIBPfWh43SZSGAB1WPRkAA4ABlcriAzb3N0Dq7h2ip4kIAI6sDBCLoMHp9JCpAauDIEDCVaq1LDlfRwACKrAg8G2rhgx3uNEeSA2RNyGEM5AAwoRegcoNBFq5WBkACrHCLwjIAX35QA